lib/ansible/module_utils/facts/hardware/base.py

```diff
@@ -0,0 +1,71 @@
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import platform
+
+from ansible.module_utils.basic import get_all_subclasses
+from ansible.module_utils.six import PY3
+
+from ansible.module_utils.facts.facts import Facts
+
+from ansible.module_utils.facts.collector import BaseFactCollector
+
+
+class Hardware(Facts):
+    """
+    This is a generic Hardware subclass of Facts.  This should be further
+    subclassed to implement per platform.  If you subclass this, it
+    should define:
+    - memfree_mb
+    - memtotal_mb
+    - swapfree_mb
+    - swaptotal_mb
+    - processor (a list)
+    - processor_cores
+    - processor_count
+
+    All subclasses MUST define platform.
+    """
+    platform = 'Generic'
+
+    def __new__(cls, *arguments, **keyword):
```

Comment:
This hackery with __new__ was something I always desired to get rid of with a facts refactor but I'll understand if you don't want to touch that until later.